pilot/pkg/features/pilot.go

```diff
@@ -174,6 +174,9 @@ var (
 	MulticlusterHeadlessEnabled = env.Register("ENABLE_MULTICLUSTER_HEADLESS", true,
 		"If true, the DNS name table for a headless service will resolve to same-network endpoints in any cluster.").Get()
 
+	IncludeAllServicesInDNS = env.Register("INCLUDE_ALL_SERVICES_IN_DNS", true,
```

Comment:
For my context, what's the use case for this?

Reply:
This is needed for DNS proxy at Gateway and forward requests using DynamicForwardProxy. This allows us to resolve unexported services as well

Reply:
Ahh ok, cool. Thanks for clarifying!